### PR TITLE
Circleci: Update nightly trivy scan

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR Description
Update CircleCI dockerScan (trivy) nightly task. Use circleci base image for more up-to-date tools. Scan both amd64 and arm64 variants of web3signer develop tag. 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

## Testing

- [x] I thought about testing these changes in a realistic/non-local environment.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -312,24 +312,16 @@ jobs:
...
           name: Scan with trivy
           shell: /bin/sh
           command: |
-            rm -rf docker/test*
-            for FILE in $(ls docker)
-            do
-              docker pull -q "consensys/web3signer:develop-$FILE"
-              trivy -q image --exit-code 1 --no-progress --severity HIGH,CRITICAL --ignorefile "gradle/trivyignore.txt" --timeout 10m "consensys/web3signer:develop-$FILE" 
-            done
+            $HOME/bin/trivy image consensys/web3signer:develop-arm64 --image-src remote --exit-code 1 --quiet --no-progress --severity HIGH,CRITICAL --ignorefile "gradle/trivyignore.txt" --timeout 10m
```
Comment (jframe) [inline .circleci/config.yml:323]: Why don't we need to do the docker pull anymore? Does the --image-src remote on trivy take care of that?

File: .circleci/config.yml [hunk 2]
```diff
@@ -312,24 +312,16 @@ jobs:
...
-            curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /usr/local/bin
-      # remove the test files and only use the folder names
+          command: curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b $HOME/bin
+      # scan both develop-arm64 and develop-amd64 images using remote src.
       - run:
           name: Scan with trivy
           shell: /bin/sh
           command: |
-            rm -rf docker/test*
-            for FILE in $(ls docker)
```
Comment (siladu) [inline .circleci/config.yml:328]: What were the previous values of $FILE?

File: .circleci/config.yml (hunk 1 above)
Comment (usmansaleem) [reply .circleci/config.yml:323]: correct. Technically we don't need to specify the `--image-src remote` as trivy tries docker, podman, containerd and lastly remote, since I was removing `setup_remote_docker`, I explicitly set it to remote so that it always fetch the image from docker hub.

File: .circleci/config.yml (hunk 2 above)
Comment (usmansaleem) [reply .circleci/config.yml:328]: we used to have `docker/jdk17/Dockerfile` and `docker/jdk11/Dockerfile`. Hence we used to create/publish `consensys/web3signer:develop-java11` and `consensys/web3signer:develop-java17`. Since we now default to java17 variant, the `develop` tag now points to `develop-java17`. In addition, we further publish platform specific variants as well i.e. `develop-amd64` and `develop-arm64`. Similarly, `develop-java17-amd64` and `develop-java17-arm64` 

In summary, we only need to test/scan `develop` tag unless we introduce `java21` tag along with `java17` tag. We were also only scanning `develop-amd64` variant in past as CircleCI trivy job runs on Linux amd64 environment. Now we make sure to scan both amd64 and arm64 variants (as both of them are built separately).

**Review by jframe — APPROVED** [review]

**Review by siladu — APPROVED** [review]
